mirrord-config/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+#![feature(slice_concat_trait)]
+#![feature(once_cell)]
+
+pub mod agent;
+pub mod config;
+pub mod env;
+pub mod feature;
+pub mod fs;
+pub mod incoming;
+pub mod network;
+pub mod outgoing;
+pub mod pod;
+pub mod util;
+
+use std::path::Path;
+
+use mirrord_config_derive::MirrordConfig;
+use serde::Deserialize;
+
+use crate::{
+    agent::AgentFileConfig, config::source::MirrordConfigSource, feature::FeatureFileConfig,
+    pod::PodFileConfig, util::VecOrSingle,
+};
+
+#[derive(MirrordConfig, Deserialize, Default, PartialEq, Eq, Clone, Debug)]
+#[serde(deny_unknown_fields)]
+#[config(map_to = LayerConfig)]
+pub struct LayerFileConfig {
+    #[config(env = "MIRRORD_ACCEPT_INVALID_CERTIFICATES", default = "false")]
+    pub accept_invalid_certificates: Option<bool>,
+
+    #[config(env = "MIRRORD_SKIP_PROCESSES")]
+    pub skip_processes: Option<VecOrSingle<String>>,
+
+    #[serde(default)]
```

Comment:
Why do you need serde(default) here?

Reply:
To not use Option for these fields but to use the Default::default from the underlying struct because it implements `MirrordConfig`

Reply:
So this is going to be mandatory for any nested field which isn't Toggleable, right? But there's no way to add it as part of the config macro?

Reply:
For fields that implement `MirrordConfig` and `Default` no need to make them optional, I removed the `Option` and added the `#[serde(default)]` for all nested fields no only the ones without Toggleable